### ISSUE TITLE
Not able to run my collection after updating postman to 5.5.0 (Collection was running fine at earlier version = 5.3.2)

### ISSUE THREAD

**Issue by maissom-abbasi-confiz** [a85/POSTMan-Chrome-Extension#3969]
## App Details:

```
postman for Windows
Version 5.5.0
win32 10.0.15063 / x64
```

## Issue Report:
 1. Did you encounter this recently, or has this bug always been there: 

My team members are encountering this issue after the update 5.5.0

 2. Expected behaviour: 

The collection should run as it was running on earlier versions of postman

 3. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):

"TypeError: Cannot read property 'toString' of undefined
    at PostmanVariable.toString (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\variable.js:108:34)
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:221:53
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:47:23
    at RegExp.[Symbol.replace] (<anonymous>)
    at String.replace (native)
    at SuperString.replace (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:45:34)
    at Substitutor.parse (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:132:27)
    at Function.replaceSubstitutions (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:222:69)
    at Function.<anonymous> (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:253:25)
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:247:40
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:4944:15
    at baseForOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:3001:24)
    at Function.forOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:13002:24)
    at Function.<anonymous> (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:246:19)
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:247:40
    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:4944:15
    at baseForOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:3001:24)"
bjectmessage: "Cannot read property 'toString' of undefined"name: "TypeError"stack: "TypeError: Cannot read property 'toString' of undefined↵    at PostmanVariable.toString (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\variable.js:108:34)↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:221:53↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:47:23↵    at RegExp.[Symbol.replace] (<anonymous>)↵    at String.replace (native)↵    at SuperString.replace (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:45:34)↵    at Substitutor.parse (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\superstring\index.js:132:27)↵    at Function.replaceSubstitutions (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:222:69)↵    at Function.<anonymous> (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:253:25)↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:247:40↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:4944:15↵    at baseForOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:3001:24)↵    at Function.forOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:13002:24)↵    at Function.<anonymous> (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:246:19)↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\lib\collection\property.js:247:40↵    at C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:4944:15↵    at baseForOwn (C:\Users\bl4\AppData\Local\Postman\app-5.5.0\resources\app.asar\node_modules\postman-collection\node_modules\lodash\lodash.js:3001:24)"

Steps to reproduce the problem
1. Open postman App
2. Go to Runner tab
3. Select desired collection (the collection contains script in both pre-request and tests tabs)
4. Select desired environment
5. Select input (file which is in csv format)
6. Hit Run Collection button.
Note: The exact above steps work perfectly fine for the same collection when run via postman version 5.3.2

![postman5 5 0-run collection error](https://user-images.githubusercontent.com/8590728/33994600-010161ca-e0fd-11e7-9c9b-b87baccd8d92.png)

**Comment by numaanashraf** [comment]
@maissom-abbasi-confiz Thanks for reporting. We are looking into this.

**Comment by harryi3t** [comment]
@maissom-abbasi-confiz If possible, could you share the collection, environment and CSV file in which you are having this issue? This would really help us in reproducing and fixing the issue.
You can remove all extra requests, tests, variables and replace endpoints to fake ones until you are able to see the error.

**Comment by maissom-abbasi-confiz** [comment]
@harryi3t 

Thanks for following up. I will update you after mocking the scenario.

**Comment by maissom-abbasi-confiz** [comment]
Hi @harryi3t ,

I spent sometime in identifying the root cause. Here is what I found and did:

There was a code block where a value was not initialized, and later being dereferenced (as an index of a specific JSON array). When I fixed this code block, there was no collection run exception and my execution (pre-request and tests scripts) ran as expected.

In earlier versions of postman ( < 5.5.0), the same code block kept running (even though it was a potential memory leak) and did not cause any exception. Anyways, a blessing in disguise :).

I assume the postman labs team could have done some optimizations related to such areas and rightly caught the exception causing area.

One observation though:
The exception message did not lead me directly to the buggy area. I would put a console.log at the end of pre-request script and it would still get printed, making me thing that the pre-request script is running fine. On the contrary, the buggy code was identified inside the pre-request script.

Is it possible, that such exceptions are thrown right when they occur, and there is meaningful information (such as line number and exact memory reference that causes error/exception). In my case, I was not doing ".toString" in pre-request script anywhere, but the exception message (as shown in the screenshot) had message related to that.

Thanks.
